bindata/network/openshift-sdn/001-crd.yaml

```diff
@@ -1,207 +1,326 @@
+---
```

Comment:
looks like the generated CRDs are missing range validations and `additionalPrinterColumns` as well as comment text around the `validation` stanza. If we need to add those back, I can merge the changes from the old CRD yaml into this.

Reply:
We shouldn't have two copies; if there's a generated copy in library-go we should update the code to pull from that directly.

As for the differences:
- We can live without the comments
- The loss of validation on `ClusterNetwork` isn't that big of a deal since it's not user-generated anyway. (Also, if we end up merging any of the openshift-sdn IPv6 work, we'd have to drop the IP/CIDR validation anyway.)
- The loss of validation on the "egress" fields in `NetNamespace` and `HostSubnet` is a little more of a problem, since those are user-modifiable. Likewise all of `EgressNetworkPolicy`. If we lose validation there we need to make sure that the code in openshift-sdn validates the fields when it reads them in and won't crash if they're bad.
  - Alternately, maybe we should extend multus-admission-controller to be cno-admission-controller and validate SDN types too?
  - Alternately you may be able to add back some of the validation by adding `kubebuilder` annotations to the API objects in `openshift/api`.
- Likewise I think you can get the `additionalPrinterColumns` back via `kubebuilder` annotations and if so we should definitely do that because `oc get` becomes useless for these types without that.

Reply:
Fair enough. I will try to add `kubebuilder` annotations/markers for `additionalPrinterColumns` as well as field validations where possible. For the other ones, I will merge the existing ones in to the generated CRDs. There are no two copies of it - just the one checked in under `bindata/network`. There is none under `openshift/api` repo. I was asking @alexanderConstantinescu how we generated the `001-crd.yaml` to begin with. Looking at the file history @squeed checked in that yaml, but looks like it was a manual process most likely.

Reply:
Is it possible for this to just be a symlink in the `manifests` directory? That way we more easily grasp what is installed.

Yes, we should be adding kubebuilder annotations to `openshift/api` as much as possible.

Reply:
@squeed Not sure I understood the first part of your comment. The `001-crd.yaml` doesn't exist at all in the manifests directory today. Do you want me to add a sym-link to this file in `manifests`? Or are you referring to the `0000_70_cluster-network-operator_01_crd.yaml` which is currently checked into `vendor/` as well as has a copy in the `manifests` dir?